### PR TITLE
Adding RoundUp

### PR DESCRIPTION
**Pull request by da0x**
Adding round up. Rounds decimal towards positive infinite.

### CODE REVIEW

**Comment by mwoss** [comment]
Thanks, @da0x for your contribution. I will review ur PR tomorrow. I have to communicate with Jason (second contributor) as I know he has already started the implementation of RoundUp/Down methods.

File: decimal.go [hunk 1]
```diff
@@ -902,6 +902,18 @@ func (d Decimal) RoundBank(places int32) Decimal {
 	return round
 }
 
+// RoundUp rounds decimal number towards positive infinite
+// (1.1).RoundUp(0) => 1
+// (1.52).RoundUp(1) => 1.6
+// (1.1001).RoundUp(2) => 1.11
+func (d Decimal) RoundUp(places int32) Decimal {
+	factor := NewFromFloat(10).Pow(NewFromInt(int64(places)))
```
Comment (lovung) [inline decimal.go:910]: Decimal Pow is not a high-performance method. I have benchmarked them before but not have results here. You can try to benchmark it.

How about 
```go
factor := NewFromFloat(math.Pow(10, float64(-places))
```

Comment (da0x) [reply decimal.go:910]: Thank you! Great suggestion, I'll try it out.

Comment (da0x) [reply decimal.go:910]: using math.Pow:
```
BenchmarkDecimal_RoundUp-8                            	  799142	      1522 ns/op
```
using Decimal.Pow
```
BenchmarkDecimal_RoundUp-8                            	  255872	      4411 ns/op
```

I'll update the PR with your suggestion.

**Comment by mwoss** [comment]
I was thinking for a bit about your implementation and I guess there is still a place for improvement. As we decided with Jason to include RoundUp/Down methods in the 1.3 release I would reiterate the implementation to be more performant. 
I have already prepared the improved implementation for RoundUp, I think I will finish the implementation of RundDown tomorrow (don't have much time today tho :<)
We can collaborate on creating tests for those methods if you want :DD 

Thanks a lot for your contribution and interest in decimal lib! :3

**Comment by da0x** [comment]
I agree, N*10^n can be easily optimized by shifting the decimal point instead of multiply/divide. I'm ok with this implementation though since performance isn't an issue for me. You can accept this PR and later replace it with your implementation or just keep it as a reference for others until yours is deployed. Up to you! Thanks for building this library guys. very useful.

**Review by lovung — APPROVED** [review]
LGTM. Thank you so much.

**Comment by lovung** [comment]
Related with this issue https://github.com/shopspring/decimal/issues/191

**Comment by mwoss** [comment]
I apologize I still did not create PR with my solution. I had a lot of things to do today. I promise I will do it tomorrow, I will keep you posted :))

**Comment by da0x** [comment]
Resolved: https://github.com/shopspring/decimal/pull/196#issue-536383391

**Pull request closed without merging**